src/action/action.ts

```diff
@@ -0,0 +1,440 @@
+
+    // retrieve the schema and run validations
+    const { dbid, actionName, preparedActions, modifiers } = await this.checkSchema(
+      cached as ActionInput[]
+    );
+
+    // throw runtime error if signer is null or undefined
+    const { signer, chainId, identifier, signatureType } = objects.validateFields(
+      {
+        signer: this.signer,
+        chainId: this.chainId,
+        identifier: this.identifier,
+        signatureType: this.signatureType,
+      },
+      (fieldName: string) => `${fieldName} is required to build a transaction.`
+    );
+
+    // throw runtime error if action is a view action. view actions require a different payload structure than transactions.
+    if (modifiers && modifiers.includes('VIEW')) {
+      throw new Error(
+        `Action / Procedure ${actionName} is a 'view' action. Please use kwil.call().`
+      );
+    }
+
+    // construct payload
+    const payload: UnencodedActionPayload<PayloadType.EXECUTE_ACTION> = {
+      dbid: dbid,
+      action: actionName,
+      arguments: preparedActions,
+    };
+
+    // build the transaction
+    return await Payload.createTx(this.kwil, {
+      payloadType: PayloadType.EXECUTE_ACTION,
+      payload: payload,
+      signer: signer,
+      signatureType: signatureType,
+      description: this.description,
+      challenge: this.challenge,
+      signature: this.signature,
+      chainId: chainId,
+      identifier: identifier as Uint8Array, // Fix this. Seems hacky...
+      nonce: this.nonce,
+    })
+      .buildTx()
+      .finally(() => (this.actionInputs = cached));
+  }
+
+  /**
+   * Builds the message structure for view actions. This can be provided to the `kwil.call()` api.
+   */
+  async buildMsg(): Promise<Message> {
+    this.assertNotBuilding();
+
+    // cache the action
+    const cached = this.actionInputs;
+
+    // set the actions to a special value to indicate that the message is being built
+    this.actionInputs = TXN_BUILD_IN_PROGRESS;
+
+    // retrieve the schema and run validations
+    const { dbid, actionName, preparedActions, modifiers } = await this.checkSchema(
+      cached as ActionInput[]
+    );
+
+    // throw a runtime error if more than one set of inputs is trying to be executed. Call does not allow bulk execution.
+    if (preparedActions && preparedActions.length > 1) {
+      throw new Error(
+        'Cannot pass more than one input to the call endpoint. Please pass only one input and try again.'
+      );
+    }
+
+    // throw runtime error if action is not a view action. transactions require a different payload structure than view actions.
+    if (modifiers && modifiers.includes('VIEW') === false) {
+      throw new Error(`Action ${actionName} is not a view only action. Please use kwil.execute().`);
+    }
+
+    // resolve if a signer is or is not specified. Only actions with `mustsign` require a signer.
+    const signer = this.signer ? this.signer : null;
```

Comment:
`mustsign` is an old thing. We can prob just remove this and pass `this.signer` to `Payload.createTx`.